### PR TITLE
WIP: feat: add general app logic & setup of gateway configuration section

### PR DESCRIPTION
**Pull request by Mohammer5**
This will add a basic frame (which still needs styling) and routing mechanism

### CODE REVIEW

File: src/sidebar/SidebarItem.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { useHistory, useLocation } from 'react-router-dom'
+import { MenuItem } from '@dhis2/ui'
+import { PropTypes } from '@dhis2/prop-types'
+import React from 'react'
+
+export const SidebarItem = ({ label, path }) => {
+    const history = useHistory()
+    const location = useLocation()
```
Comment (HendrikThePendric) [inline src/sidebar/SidebarItem.js:8]: You could use a slightly different hook:
```javascript
const MenuItemLink = ({ to, label }) => {
    const history = useHistory()
    const match = useRouteMatch(to)
    const onClick = () => history.push(to)
    return <MenuItem label={label} active={!!match} onClick={onClick} />
}
```

Comment (Mohammer5) [reply src/sidebar/SidebarItem.js]: Done, thanks

**Comment by amcgee** [comment]
@Mohammer5 this is a huge PR!  Is it ready for review or still in-progress?  If ready, where should someone start in reviewing it?  Is @HendrikThePendric already on the case?

**Comment by ismay** [comment]
I'll explain the approval a bit, for those not in the meeting @HendrikThePendric, @Mohammer5 and I just had. We're under a _very_ tight deadline for this app, and since Hendrik and I want to start work on this app as well, we've decided to merge this to master and branch off of there. Not ideal, but we'll have to move quick if we want to make it.

**Comment by Mohammer5** [comment]
Thanks for explaining @ismay and sorry @amcgee that I haven't reacted to your commect..

**Pull request merged**